lints/lint_subject_common_name_not_from_san.go

```diff
@@ -24,16 +26,16 @@ func (l *subjectCommonNameNotFromSAN) CheckApplies(c *x509.Certificate) bool {
 }
 
 func (l *subjectCommonNameNotFromSAN) Execute(c *x509.Certificate) *LintResult {
-	cn := c.Subject.CommonName
+	cnLowerCase := strings.ToLower(c.Subject.CommonName)
 
 	for _, dn := range c.DNSNames {
-		if cn == dn {
+		if cnLowerCase == dn {
```

Comment:
Are we sure that DNSNames will always be lower case?

Reply:
Actually, no. That's a good point. I haven't seen capitalised domain name in the SAN DNS in the wild, and I had a look at the relevant RFCs ( IDNA related ) and could not find any requirements for having the domain names in lower case. I have found some information leading me to the conclusion that it should not matter. 

I will update the PR. 
Also, while going through the RFCs I have noticed this passage in the rfc 3490: 
`The ACE prefix for IDNA is "xn--" or any capitalization thereof.`

This is not how the `e_international_dns_name_not_unicode` is working. 
```
[...]
if strings.HasPrefix(label, "xn--") {
	_, err := idna.ToUnicode(label)
	if err != nil {
		return &LintResult{Status: Error}
	}
}
[...]
```
This test is case sensitive.  What i have also found is that golang's idna package doesn't handle this case properly either. 
```
XN--FIQS8S.ICOM.MUSEUM -> XN--FIQS8S.ICOM.MUSEUM [no error]
xn--fiqs8s.icom.museum -> 中国.icom.museum [no error]
```
I think it warrants another PR for zlint. 

I would have to investigate if the golang package doesn't have a bug. I will take a working assumption that I am missing something, bug is rather unlikely. Chrome and curl  handles XN--FIQS8S.ICOM.MUSEUM properly. 
Curl sends Host: `XN--FIQS8S.ICOM.MUSEUM` header so the webserver is handling the case properly also ( this excludes any mangling by Chrome itself. )